src/bootstrap/download.rs

```diff
@@ -318,15 +318,27 @@ impl Config {
         let channel = format!("{version}-{date}");
 
         let host = self.build;
-        let rustfmt_path = self.initial_rustc.with_file_name(exe("rustfmt", host));
-        let bin_root = self.out.join(host.triple).join("stage0");
+        let bin_root = self.out.join(host.triple).join("rustfmt");
+        let rustfmt_path = bin_root.join("bin").join(exe("rustfmt", host));
```

Comment:
This broke the rust-analyzer config we suggest in the dev-guide: https://rustc-dev-guide.rust-lang.org/building/suggested.html#configuring-rust-analyzer-for-rustc
```json
    "rust-analyzer.rustfmt.overrideCommand": [
        "./build/host/stage0/bin/rustfmt",
        "--edition=2021"
    ],
```
since now rustfmt is in `build/host/rustfmt/bin/rustfmt` instead of `build/host/stage0/bin/rustfmt`. What was the reason for this change? Is it possible to revert it?

Reply:
I think it's because we can't install multiple `rustc` components to the same prefix path. Maybe we can symlink `rustfmt` in the old place, although that won't work for Windows. Better would still be to use the same `rustfmt` as the bootstrap compiler, if we can figure out the unstable features for that. (or stabilize what we need!)

Reply:
See https://github.com/rust-lang/rust/issues/107547 for more discussion.